src/types.rs

```diff
@@ -329,15 +327,15 @@ mod test {
         use super::DateTime;
         let dt = DateTime::default();
         assert_eq!(dt.timepart(), 0);
-        assert_eq!(dt.datepart(), 0b0000000_0001_00001);
+        assert_eq!(dt.datepart(), 0b0000_0000_0010_0001);
```

Comment:
The formatting here is useful: `datepart` has three bitfields with different lengths, so using runs of 4 is confusing. (That's the first of January. Not so obvious with the new formatting ;))

Reply:
Fixed